public/script.js

```diff
@@ -17,3 +17,72 @@ if ("serviceWorker" in navigator) {
       console.error(err);
     });
 }
+
+function userAgentDetectDevice(){
+  // if the userAgent contains "Windows"
+  if(navigator.userAgent.indexOf("Windows") !== -1){
+    return "Windows"
+  }
+  // if the userAgent contains "Macintosh"
+  else if(navigator.userAgent.indexOf("Macintosh") !== -1){
+    return "macOS"
+  }
+  // if the userAgent contains "Linux"
+  else if(navigator.userAgent.indexOf("Linux") !== -1){
+    return "Linux"
+  }
+  // if the userAgent contains "Android"
+  else if(navigator.userAgent.indexOf("Android") !== -1){
+    return "Android"
+  }
+  // if the userAgent contains "iPhone" or "iPad" or "iPod"
+  else if(navigator.userAgent.indexOf("iPhone") !== -1 || navigator.userAgent.indexOf("iPad") !== -1 || navigator.userAgent.indexOf("iPod") !== -1){
+    return "iOS"
+  }
+}
+
+function detectDevice(){
+  // try to detect using navigator.userAgentData.platform
+  try{
+    // windows
+    if(navigator.userAgentData.platform === "Windows"){
+      return "Windows"
+    }
+    // macOS
+    else if(navigator.userAgentData.platform === "MacIntel" || navigator.userAgentData.platform === "macOS"){
+      return "macOS"
+    }
+    // linux
+    else if(navigator.userAgentData.platform === "Linux x86_64"){
+      return "Linux"
+    }
+    // android
+    else if(navigator.userAgentData.platform === "Android"){
+      return "Android"
+    }
+    // ios
+    else if(navigator.userAgentData.platform === "iPhone" || navigator.userAgentData.platform === "iPad" || navigator.userAgentData.platform === "iPod"){
+      return "iOS"
+    }else{
+      console.log("navigator.userAgentData.platform didn't return a supported value")
+      return userAgentDetectDevice()
+    }
+  }catch(e){
+    console.log("navigator.userAgentData.platform not supported")
+    return userAgentDetectDevice()
+  }
+}
+
+// if detectDeviceMessage element exists
+if (document.getElementById("detectDeviceMessage")) {
+  // Detect device
+  const device = detectDevice();
+  if(device === ""){
+    console.log("detectDevice() retruned an empty string")
```

Comment:
Small spelling mistake

```suggestion
    console.log("detectDevice() returned an empty string")
```